src/mcpdotnet/Client/McpClientFactory.cs

```diff
@@ -43,10 +44,22 @@ public class McpClientFactory : IDisposable
     public McpClientFactory(
         IEnumerable<McpServerConfig> serverConfigs,
         McpClientOptions clientOptions,
-        ILoggerFactory loggerFactory,
+        ILoggerFactory? loggerFactory = null,
         Func<McpServerConfig, IClientTransport>? transportFactoryMethod = null,
         Func<IClientTransport, McpServerConfig, McpClientOptions, IMcpClient>? clientFactoryMethod = null)
     {
+        if (serverConfigs is null)
+        {
+            throw new ArgumentNullException(nameof(serverConfigs));
+        }
+
+        if (clientOptions is null)
+        {
+            throw new ArgumentNullException(nameof(clientOptions));
+        }
+
+        loggerFactory ??= NullLoggerFactory.Instance;
```

Comment:
Does this serve to address the issue with logs interfering with stdio transports?

Reply:
Partially. This won't provide a problematic logger, but one could still be picked up from DI I believe based on how things are structured, so if you put a consoler logger into DI, depending on how things are structured you might still have issues.

Reply:
What's would be a long-term solution? Replace DI with explicit parameterization? Remove logging from the components altogether? It seems the first approach removes the obvious footgun from novice users but we can't entirely prevent app components writing to console.

Reply:
A stdio-based protocol is fraught with peril. I can't think of a perfect solution. If we integrate nicely with DI and someone uses a console logger, it breaks. If we don't integrate nicely with DI, then we lose out on any other more production ready form of logging that's been configured. Docs? Analyzers? Not sure.

Reply:
Is it possible to introspect the logger instance and throw an exception if it's found to be attaching a known console logger type?